### PR TITLE
Rename pilot hubs to infrastructure

### PR DESCRIPTION
**Pull request by choldgraf**
This renames references of `pilot hubs` to `infrastructure` or `community hubs` when we refer to specific hubs.

### CODE REVIEW

**Comment by damianavila** [comment]
General comment, you are making some changes in our templates to fix comment links...
I would be hesitant to do that because we are going to trigger unneeded deployments just to fix a comment link.
I would prefer to change those at the time we implement change in the terraform files, IMHO.

**Comment by choldgraf** [comment]
Resolved conflicts! I wonder if @damianavila would be willing to take one last look to make sure I'm not overwriting something important?

File: docs/structure.md [hunk 1]
```diff
@@ -79,8 +79,8 @@ repository, available at [2i2c.org/pilot](https://2i2c.org/pilot/)
 
 These are folks tasked with building, maintaining, debugging and fixing
 2i2c infrastructure. Documentation targetting them should be in
-[2i2c-org/pilot-hubs](https://github.com/2i2c-org/pilot-hubs).
-repository (2i2c-org/pilot-hubs), regardless of the kind of hub it 
+[2i2c-org/infrastructure](https://github.com/2i2c-org/infrastructure).
+repository (2i2c-org/infrastructure), regardless of the kind of hub it
```
Comment (sgibson91) [inline docs/structure.md:83]:
```suggestion
[2i2c-org/infrastructure](https://github.com/2i2c-org/infrastructure)
repository (2i2c-org/infrastructure), regardless of the kind of hub it 
```

**Comment by sgibson91** [comment]
For future reference, the path list the would cause new deployments can be found here:

https://github.com/2i2c-org/infrastructure/blob/be6c30f5ce31f49081c6426fdcbac712137110f5/.github/workflows/deploy-hubs.yaml#L7-L15

I left a couple of small comments, otherwise it looks good.

File: docs/topic/terraform.md [hunk 2]
```diff
@@ -71,7 +71,7 @@ terraform init -backend-config=backends/<REQUIRED_CONFIG>.hcl -reconfigure
 terraform workspace select WORKSPACE_NAME
 ```
 
-For example, if you were working on our `pilot-hubs`, with our default backend initialised, but wanted to switch to working on our Pangeo deployments, the commands would look as follows:
+For example, if you were working on our `infrastructure`, with our default backend initialised, but wanted to switch to working on our Pangeo deployments, the commands would look as follows:
```
Comment (choldgraf) [inline docs/topic/terraform.md:74]:
```suggestion
For example, if you were working on our `pilot-hubs`, with our default backend initialised, but wanted to switch to working on our Pangeo deployments, the commands would look as follows:
```

**Comment by damianavila** [comment]
> Resolved conflicts! I wonder if @damianavila would be willing to take one last look to make sure I'm not overwriting something important?

I was willing but people have beaten me at it 😜  .

**Pull request merged**